### PR TITLE
[WIP] Closes #1869: Alphabetical filter and search view for person

### LINKED ISSUE #1869: Alphabetical filter and search view for person
**Do not disclose security vulnerabilities here, please email us instead: 
az-digital-security@list.arizona.edu**

## Motivation
I want to display people in a row view with alphabetical groupings and search by name filter.

### Is your feature request related to a problem? Please describe.
Alpha index view does not allow for person row display.

## Proposed Resolution
Add 2 new view displays to person view.

### Describe alternatives you've considered
Updating alpha index view won't work because that template is hard-coded.

### Roles and Permissions considerations
A clear and concise description of how each of the following roles would be impacted by this change:
- Content editor
- Content administrator
- Administrator

## Additional context
- Content editor workaround: https://live-azs-optics.pantheonsite.io/osc-arizona/directory
- Custom view: https://live-azs-psychology.pantheonsite.io/faculty

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description
- Enables and configures az_alphabetical_listing module by default
- Makes az_person dependent on az_alphabetical_listing module
- Adds 4 new view displays to existing AZ person view
  - Grid with Search
  - Row with Search
  - Grid with Search and Alphabetical Filter
  - Row with Search and Alphabetical Filter

## Related issues
Closes #1869 

## How to test
1. Add person content
  a. Ensure Alphabetical Index Letter field is configurable
2. Select new views to add to pages
  a. _Note: Having search and search/filter views on one page will mess with filter styles a bit._ 

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by camikazegreen** [comment]
The Alphabetical Index Letter has been added to the person content type, but has not been included in the form display, so there is currently no way for Content Admins to assign an index letter.

When the Alpha filter was originally added, it was made to use the index letter so that it could be very generic and be added to many different content types. If we are adding custom views for People specifically, I would recommend not making it dependent on index letter at all, and using Last Name, First Name sorting instead.

**Comment by tadean** [comment]
> The Alphabetical Index Letter has been added to the person content type, but has not been included in the form display, so there is currently no way for Content Admins to assign an index letter.
> 
> When the Alpha filter was originally added, it was made to use the index letter so that it could be very generic and be added to many different content types. If we are adding custom views for People specifically, I would recommend not making it dependent on index letter at all, and using Last Name, First Name sorting instead.

One discussion we'd had about this in Drupal 7 was that it didn't handle the edge case of people with accent marks in their name; they'd end up in a category by themselves that wasn't a normal alphabet letter. At the time, the index letter field was considered to possibly be a fix for that.

**Comment by camikazegreen** [comment]
Was it this guy: Grimme Ælling

**Comment by danahertzberg** [comment]
> The Alphabetical Index Letter has been added to the person content type, but has not been included in the form display.

Halp please! I've done this many times on my local, but I'm not finding the right file to export apparently. I would like the field to come right after Last Name please. 🙏

File: modules/custom/az_person/config/install/core.entity_form_display.node.az_person.default.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ status: true
 dependencies:
```
Comment (trackleft) [inline modules/custom/az_person/config/install/core.entity_form_display.node.az_person.default.yml:3]: Put this file unchanged here: https://github.com/az-digital/az_quickstart/tree/main/modules/custom/az_seo/config/quickstart

Comment (trackleft) [reply modules/custom/az_person/config/install/core.entity_form_display.node.az_person.default.yml:3]: Then add the same file here without lines:

- https://github.com/az-digital/az_quickstart/pull/1871/files#diff-ec610764f264f1365ff8044c366354a9064ec1b795dd50c10fb1f36672e495e5R16
- https://github.com/az-digital/az_quickstart/pull/1871/files#diff-ec610764f264f1365ff8044c366354a9064ec1b795dd50c10fb1f36672e495e5R29
- https://github.com/az-digital/az_quickstart/pull/1871/files#diff-ec610764f264f1365ff8044c366354a9064ec1b795dd50c10fb1f36672e495e5R163-R170

Comment (trackleft) [reply modules/custom/az_person/config/install/core.entity_form_display.node.az_person.default.yml:3]: That should resolve the issue.

Comment (trackleft) [reply modules/custom/az_person/config/install/core.entity_form_display.node.az_person.default.yml:3]: You need the same file twice, once with the metatags, once without.

Comment (trackleft) [reply modules/custom/az_person/config/install/core.entity_form_display.node.az_person.default.yml:3]: But of course, that means that we need to make az_person a dependency on az_seo (probably)

**Comment by trackleft** [comment]
Maybe we should look into config actions for this type of dependency issue https://www.drupal.org/project/config_actions

**Comment by danahertzberg** [comment]
https://github.com/az-digital/az_quickstart/issues/985#issuecomment-940501793
> Follow-up improvements:
>
> Investigate implementing the built in "glossary" functionality/display (to eliminate using JS navigation)
Investigate the issue with search field not displaying in the view (to eliminate using JS search)
Investigate improving the alpha nav experience on mobile (dropdown?)
Fix anchor link functionality on mobile (alpha nav overlapping group title). Also investigate jumpiness on click on nav item (desktop & mobile).

**Comment by danahertzberg** [comment]
Can also trim last name field to be used as results grouping

**Pull request closed without merging**